dali/test/argument_key_test.cc

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <dali/test/argument_key.h>
+#include <gtest/gtest.h>
+
+namespace dali {
+namespace testing {
+
+TEST(ArgumentKeyTest, ArgumentKeyTest) {
+
+  ArgumentKey argk("arg");
+  EXPECT_EQ("arg", argk.arg_name());
+  EXPECT_EQ("", argk.node_name());
+
+  ArgumentKey opargk("op", "arg");
+  EXPECT_EQ("arg", opargk.arg_name());
+  EXPECT_EQ("op", opargk.node_name());
+}
+
+
+TEST(ArgumentKeyTest, ArgumentOperatorLessTest) {
+  ArgumentKey a1("arg1");
```

Comment:
Add test(s) with compound keys, preferably with first and second part equal:
`ArgumentKey a1("A", "X"), a2("A", "Y"), a3("B", "X");`
then compare `a1 < a2` and `a1 < a3'.

Reply:
Done